### PR TITLE
More more tests

### PR DESCRIPTION
**Pull request by Licenser**

### CODE REVIEW

**Comment by Licenser** [comment]
This is quite odd, it seems that when running with tarpaulin we manage to segfault o.O

**Comment by Licenser** [comment]
The cause was a regression in the new 0.9 release - see https://github.com/xd009642/tarpaulin/issues/264

**Comment by Licenser** [comment]
Aside of improvements I found (and fixed) this issue: https://github.com/lemire/simdjson/issues/325

File: src/value/borrowed/cmp.rs [hunk 1]
```diff
@@ -1,101 +1,127 @@
...
+use float_cmp::approx_eq;
 
-impl<'a> PartialEq<()> for Value<'a> {
-    fn eq(&self, _other: &()) -> bool {
-        if let Value::Null = self {
-            true
-        } else {
-            false
+impl<'a> PartialEq for Value<'a> {
+    fn eq(&self, other: &Self) -> bool {
```
Comment (sunnygleason) [inline src/value/borrowed/cmp.rs]: does this automatically get `#[inline]`?

**Comment by sunnygleason** [comment]
BTW this is super impressive!

image

File: .github/workflows/validate.yaml [hunk 2]
```diff
@@ -37,11 +37,11 @@ jobs:
           toolchain: nightly
           override: true
       - name: install tarpaulin
-        run: cargo install cargo-tarpaulin -f
+        run: cargo install cargo-tarpaulin -f # --version 0.8.7
```
[the remark this answers is not in the document]
Comment (Licenser) [reply .github/workflows/validate.yaml]: Good catch!

File: src/serde/value/owned/se.rs [hunk 3]
```diff
@@ -655,3 +655,13 @@ impl serde::ser::SerializeStructVariant for SerializeStructVariant {
...
+
+#[cfg(test)]
+mod test {
+    //    use super::*;
+    //    use supper::super::*;
+
+    //    fn se_u8() {
+    //let v: Value = crate::serde::
+    //    }
+}
```
[the remark this answers is not in the document]
Comment (Licenser) [reply src/serde/value/owned/se.rs]: Was my fist attempt will fill it in another pr, deterring for niw

File: src/value/borrowed/cmp.rs (hunk 1 above)
Comment (Licenser) [reply src/value/borrowed/cmp.rs]: Ja rustc is pretty smart about inlining functions. We only use it in other places because we think we know better 😂

**Comment by sunnygleason** [comment]
:shipit:  🚀

**Comment by Licenser** [comment]
tagged and published as v0.1.22

**Pull request merged**